File: SourceCode/Angular/src/app/core/domain-classes/customer.ts
import { ContactAddress } from './contact-address';
import { EntityState } from './entity-state';

export interface Customer {
  id: string;
  customerName: string;
  contactPerson: string;
  email: string;
  mobileNo: string;
  phoneNo: string;
  objectState?: EntityState;
  isDeleted?: boolean;
  isVerify?: boolean;
  isSendMail?: boolean;
  description: string;
  website: string;
  url?: string;
  imageUrl?: string;
  logo?: string;
  isImageUpload?: boolean;
  isWalkIn?: boolean;
  taxNumber?: string;
  billingAddress?: ContactAddress;
  shippingAddress?: ContactAddress;
  billingAddressId?: string;
  shippingAddressId?: string;
}

File: SourceCode/SQLAPI/POS.Data/Entities/Customer/Customer.cs
using POS.Data.Entities;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace POS.Data
{
    public class Customer : BaseEntity
    {
        public Guid Id { get; set; }
        public string CustomerName { get; set; }
        public string ContactPerson { get; set; }
        public string Email { get; set; }
        public string Fax { get; set; }
        public string MobileNo { get; set; }
        public string PhoneNo { get; set; }
        public string Website { get; set; }
        public string Description { get; set; }
        public string Url { get; set; }
        public Guid? BillingAddressId { get; set; }
        [ForeignKey("BillingAddressId")]
        public ContactAddress BillingAddress { get; set; }
        public Guid? ShippingAddressId { get; set; }
        [ForeignKey("ShippingAddressId")]
        public ContactAddress ShippingAddress { get; set; }
        public bool IsWalkIn { get; set; }
        public string TaxNumber { get; set; }
        
        // Sales Person & Region attribution
        public Guid? SalesPersonId { get; set; }
        [ForeignKey("SalesPersonId")]
        public User SalesPerson { get; set; }
        
        public Guid? LocationId { get; set; }
        [ForeignKey("LocationId")]
        public Entities.Location Location { get; set; }

        [NotMapped]
        public string ImageUrl { get; set; }

    }
}
